test/stream_test/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+set(_IRODS_TEST_EXECUTABLES_BUILD_DOCSTR "Build and package executables made for testing.")
+
+# Building the test executables is enabled by default for Debug builds.
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+	set(_IRODS_TEST_EXECUTABLES_BUILD_DEFAULT YES)
+else()
+	set(_IRODS_TEST_EXECUTABLES_BUILD_DEFAULT NO)
+endif()
+
+set(IRODS_TEST_EXECUTABLES_BUILD ${_IRODS_TEST_EXECUTABLES_BUILD_DEFAULT} CACHE BOOL "${_IRODS_TEST_EXECUTABLES_BUILD_DOCSTR}")
+
+if (NOT IRODS_TEST_EXECUTABLES_BUILD)
+	return()
+endif()
```

Comment:
I'd like to see this sort of thing (from the top of this file to this line) in a `CMakeLists.txt` in the `test` directory. That way we won't have to move it later if we add more tests. That said, this is trivial enough that I wouldn't mind at all if you'd rather kick the can down the road.